src/App/src/Middleware/ContentNegotiationMiddleware.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Api\App\Middleware;
+
+use Dot\AnnotatedServices\Annotation\Inject;
+use Laminas\Diactoros\Response\JsonResponse;
+use Laminas\Http\Response;
+use Mezzio\Router\RouteResult;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+use function array_filter;
+use function array_intersect;
+use function array_map;
+use function explode;
+use function in_array;
+use function is_array;
+use function str_contains;
+use function strtok;
+use function trim;
+
+class ContentNegotiationMiddleware implements MiddlewareInterface
+{
+    /**
+     * @Inject({"config.content-negotiation"})
+     */
+    public function __construct(private array $config)
+    {
+    }
+
+    public function process(
+        ServerRequestInterface $request,
+        RequestHandlerInterface $handler
+    ): ResponseInterface {
+        /** @var RouteResult $routeResult */
+        $routeResult = $request->getAttribute(RouteResult::class);
+        if (! $routeResult instanceof RouteResult || $routeResult->isFailure()) {
+            return $handler->handle($request);
+        }
+
+        $routeName = $routeResult->getMatchedRouteName();
```

Comment:
```suggestion
        $routeName = (string) $routeResult->getMatchedRouteName();
```
This is mostly to stop Psalm complaining that:

> psalm: PossiblyFalseArgument: Argument 1 of Api\App\Middleware\ContentNegotiationMiddleware::checkAccept cannot be false, possibly string value expected

Because `getMatchedRouteName`'s signature includes boolean as well, we need to cast `$routeName` to string.
Casting can be done safely, as at this point we know for sure that a route was matched, which must have a name.